hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerRequestHandler.java

```diff
@@ -1218,8 +1219,18 @@ private GetS3VolumeContextResponse getS3VolumeContext()
     return impl.getS3VolumeContext().getProtobuf();
   }
 
+  @DisallowedUntilLayoutVersion(SNAPSHOT_SUPPORT)
   private SnapshotDiffResponse snapshotDiff(
       SnapshotDiffRequest snapshotDiffRequest) throws IOException {
+        if (!getOzoneManager().getVersionManager()
+        .isAllowed(SNAPSHOT_SUPPORT)) {
+          throw new OMException(
+                  "cannot be invoked before finalization.",
+                  OMException.ResultCodes.
+                    NOT_SUPPORTED_OPERATION_PRIOR_FINALIZATION)
+                  ;
+        }
+
```

Comment:
nit:
1. You can extract out this code to some validation function and reuse at  other places like. May be add it to [OmSnapshotUtils](https://github.com/apache/ozone/blob/06433c0da953d885ce39e439e0c2ddc8d96af6f6/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/OmSnapshotUtils.java#L41)
2. nit: indentation is bit off. Please configure your intelliJ with Ozone style: https://www.jetbrains.com/help/idea/configuring-code-style.html#import-code-style

Reply:
Yes good catch! Definitely we should will do so! Thanks Hemant!